release.md

```diff
@@ -0,0 +1,27 @@
+# pyodide-build release workflow
+
+This document provides maintenance and release instructions for `pyodide-build` project, adapted
+from the [Maintainer information page from the Pyodide project](https://pyodide.org/en/stable/development/maintainers.html)
+
+> [!NOTE]
+> `pyodide-build` is also used as a submodule in-tree within the Pyodide repository, and it is usually acceptable to
+update the submodule to point to a newer commit in most cases and a release isn't always required unless it's needed
+to address bugs or add features for out-of-tree builds.
+>
+> See the [Updating `pyodide-build](https://pyodide.org/en/stable/development/maintainers.html#updating-pyodide-build) for more information.
+
+# Release instructions
+
+- First, create a "Release planning" tracking issue that can list information such as:
+    - the timeline of the release
+    - the scope of the release and the pull requests to be included, ideally by attaching a milestone to them or creating one through the GitHub Issues UI if it doesn't exist
+    - any other relevant resources
+
+- Once a consensus is reached on the release plan, create a new milestone in the GitHub Issues UI. This milestone should have the same name as the release version (e.g., `v0.23.0`) and should be marked as "open". The milestone should also be assigned to the release planning issue.
```

Comment:
I'd expect this amount of process will be unnecessary. Ideally if any single maintainer feels like making a release for any reason they should be able to just do it, unless there is some compelling reason to have more process. Releases are small and they should be easy to make as long as we have faith in our CI pipelines. If something is missing, we can just make a followup release.

I think the only thing that we might need to discuss is whether a release should be major or minor. I think it would be good to have a policy here on how we decide that.

Reply:
+1.

If there isn't a breaking change, I think a maintainer can make a release when necessary. That's the main reason why we made a separate pyodide-build repository.

I think we can make an issue when we have a reason to block the release until some fixes are merged.